### PR TITLE
Fix valid ClusterId check

### PR DESCRIPTION
**Pull request by vthapar**
Current check for validating ClusterId allows `.` in clustername.
Instead it should follow Label Names validations as defined in
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names

Fixes github.com/submariner-io/lighthouse/issues/707

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

### CODE REVIEW

File: internal/cluster/cluster.go [hunk 1]
```diff
@@ -25,9 +25,9 @@ import (
 
 func IsValidID(clusterID string) error {
 	// Make sure the clusterid is a valid DNS-1123 string
-	if match, _ := regexp.MatchString("^[a-z0-9][a-z0-9.-]*[a-z0-9]$", clusterID); !match {
+	if match, _ := regexp.MatchString("^[a-z0-9][a-z0-9-]*[a-z0-9]$", clusterID); !match {
```
Comment (mkolesnik) [inline internal/cluster/cluster.go:28]: Is suggest adding a limit on the length as well, as we discussed https://github.com/submariner-io/lighthouse/issues/707#issuecomment-1068888749

**Review by mkolesnik — APPROVED** [review]
We should probably have added test but alas there aren't any tests for subctl :(

**Review by sridhargaddam — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Review by Jaanki — APPROVED** [review]

File: internal/cluster/cluster.go (hunk 1 above)
Comment (skitt) [inline internal/cluster/cluster.go:28]: Can’t we use the same check as in https://github.com/submariner-io/lighthouse/pull/709?

**Comment by skitt** [comment]
> We should probably have added test but alas there aren't any tests for subctl :(

https://github.com/submariner-io/submariner-operator/pull/1958

File: internal/cluster/cluster.go (hunk 1 above)
Comment (mkolesnik) [reply internal/cluster/cluster.go:28]: If we do (which we probably should), the main issue would be that multiple errors are returned and would need to be converted into one, where there are two separate checks - this one and the one after that.

Some flattening of the error would be required, as the `survey` expects a single `Error`

**Comment by mkolesnik** [comment]
Ideally we should backport this to the next point release 0.12.1

File: internal/cluster/cluster.go (hunk 1 above)
Comment (skitt) [reply internal/cluster/cluster.go:28]: That’s exactly what submariner-io/lighthouse#709 does. See #1958 for the operator version.

**Comment by skitt** [comment]
Merged with #1958.

**Pull request closed without merging**